numba/core/typing/templates.py

```diff
@@ -527,9 +544,9 @@ def generic(self, args, kws):
                 self._compiled_overloads[sig.args] = disp_type.get_overload(sig)
                 # store the inliner information, it's used later in the cost
                 # model function call
-                iinfo = _inline_info(ir, typemap, calltypes, sig)
-                self._inline_overloads[sig.args] = {'folded_args': folded_args,
-                                                    'iinfo': iinfo}
+            iinfo = _inline_info(ir, typemap, calltypes, sig)
+            self._inline_overloads[sig.args] = {'folded_args': folded_args,
```

Comment:
this overrides the behaviour here https://github.com/numba/numba/pull/5673/files#diff-aee5ebdb5580a36762cba728715f96d1R536-R537 and needs looking at